src/neuroconv/tools/nwb_helpers/_models/_zarr_dataset_models.py

```diff
@@ -0,0 +1,124 @@
+"""Base Pydantic models for the ZarrDatasetConfiguration."""
+from typing import Any, Dict, List, Literal, Union
+
+import numcodecs
+import zarr
+from pydantic import Field, root_validator
+
+from ._base_dataset_models import DatasetConfiguration
+
+_base_zarr_codecs = set(zarr.codec_registry.keys())
+_lossy_zarr_codecs = set(("astype", "bitround", "quantize"))
+
+# These filters do nothing for us, or are things that ought to be implemented at lower HDMF levels
+# or indirectly using HDMF data structures
+_excluded_zarr_codecs = set(
+    (
+        "json2",  # no data savings
+        "pickle",  # no data savings
+        "vlen-utf8",  # enforced by HDMF
+        "vlen-array",  # enforced by HDMF
+        "vlen-bytes",  # enforced by HDMF
+        "msgpack2",  # think more on if we want to include this for variable length string datasets
+        "adler32",  # checksum
+        "crc32",  # checksum
+        "fixedscaleoffset",  # enforced indrectly by HDMF/PyNWB data types
+        "base64",  # unsure what this would ever be used for
+        "n5_wrapper",  # different data format
+    )
+)
+
+# Forbidding lossy codecs for now, but they could be allowed in the future with warnings?
+# (Users can always initialize and pass explicitly via code)
+_available_zarr_codecs = set(_base_zarr_codecs - _lossy_zarr_codecs - _excluded_zarr_codecs)
+
+# TODO: would like to eventually (as separate feature) add an 'auto' method to Zarr
+# to harness the wider range of potential methods that are ideal for certain dtypes or structures
+# E.g., 'packbits' for boolean (logical) VectorData columns
+# | set(("auto",))
+AVAILABLE_ZARR_COMPRESSION_METHODS = tuple(_available_zarr_codecs)
+
+
+class ZarrDatasetConfiguration(DatasetConfiguration):
+    """A data model for configuring options about an object that will become a Zarr Dataset in the file."""
+
+    # TODO: When using Pydantic v2, replace with `model_config = ConfigDict(...)`
+    class Config:
+        arbitrary_types_allowed = True
+        validate_assignment = True
+
+    compression_method: Union[Literal[AVAILABLE_ZARR_COMPRESSION_METHODS], numcodecs.abc.Codec, None] = Field(
+        default="gzip",  # TODO: would like this to be 'auto'
+        description=(
+            "The specified compression method to apply to this dataset. "
+            "Can be either a string that matches an available method on your system, "
+            "or an instantiated numcodec.Codec object."
+            "Set to `None` to disable compression."
+        ),
+    )
+    # TODO: actually provide better schematic rendering of options. Only support defaults in GUIDE for now.
+    # Looks like they'll have to be hand-typed however... Can try parsing the numpy docstrings - no annotation typing.
```

Comment:
Nah this is still a 'to-do' eventual feature perhaps; this refers to the input arguments that any individual codec would receive as 'valid'

such as 'level' of `Gzip`

But they don't use annotation typing in the signatures so that makes it harder

Reply:
Maybe hand code table for set of core supported ones but otherwise fallback to open string